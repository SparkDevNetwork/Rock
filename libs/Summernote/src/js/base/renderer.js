define(function () {
  var Renderer = function (markup, children, options, callback) {
    this.render = function ($parent) {
      var $node = $(markup);

      if (options && options.contents) {
        $node.html(options.contents);
      }

      if (options && options.className) {
        $node.addClass(options.className);
      }

      if (options && options.data) {
        $.each(options.data, function (k, v) {
          $node.attr('data-' + k, v);
        });
      }

      if (options && options.click) {
        $node.on('click', options.click);
      }

      if (children) {
        var $container = $node.find('.note-children-container');
        children.forEach(function (child) {
          child.render($container.length ? $container : $node);
        });
      }

      if (callback) {
        callback($node, options);
      }

      if (options && options.callback) {
        options.callback($node);
      }

      if ($parent) {
        $parent.append($node);
      }

      return $node;
    };
  };

  var renderer = {
    create: function (markup, callback) {
      return function () {
        var children = $.isArray(arguments[0]) ? arguments[0] : [];
        var options = typeof arguments[1] === 'object' ? arguments[1] : arguments[0];
        if (options && options.children) {
          children = options.children;
        }
        return new Renderer(markup, children, options, callback);
      };
    }
  };

  return renderer;
});
