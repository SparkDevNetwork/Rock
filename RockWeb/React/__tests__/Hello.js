
import renderer from "react-test-renderer";

import Hello from "../Hello";

it("should render Hello World", () => {
  const hello = renderer.create(<Hello />);
  expect(hello).toMatchSnapshot();
});
