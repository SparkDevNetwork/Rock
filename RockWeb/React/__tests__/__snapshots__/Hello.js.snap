exports[`test should render Hello World 1`] = `
<h1>
  Hello world
</h1>
`;
